### PR TITLE
Wasm Configuration for Ambient Mode - Test

### PR DESCRIPTION
**Pull request by adilmp1**
This PR introduces testing procedures for validating Wasm features in Ambient mode within Istio, complementing the recent documentation update (PR [#15041](https://github.com/istio/istio.io/pull/15041)). These testing guidelines aim to ensure the reliability and performance of Wasm extensions in Istio deployments.

Adds tests as per : https://github.com/istio/istio/issues/42337

### CODE REVIEW

File: content/en/docs/ambient/usage/extend-waypoint-wasm/test.sh [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+# deploy test application
+snip_deploy_the_sample_application_1
+snip_deploy_the_sample_application_2
+
+snip_deploy_the_sample_application_3
+snip_deploy_the_sample_application_4
+
+# test traffic before ambient mode is enabled
+_verify_contains snip_verify_traffic_sleep_to_ingress "$snip_verify_traffic_sleep_to_ingress_out"
```
Comment (kfaseela) [inline content/en/docs/ambient/usage/extend-waypoint-wasm/test.sh:46]: i do not see any of these verifications in the actual document.. do we need them?

Comment (kfaseela) [reply content/en/docs/ambient/usage/extend-waypoint-wasm/test.sh:46]: it is needed in the getting started document for ease of understanding, but in the wasm document i do not think we have to do this verification

File: content/en/docs/ambient/usage/extend-waypoint-wasm/test.sh [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+snip_apply_wasmplugin_at_waypoint_proxy_2
+
+# Display applied wasmplugins and verify output
+_verify_like snip_view_the_configured_wasmplugin_1 "$snip_view_the_configured_wasmplugin_1_out"
+
+# verify the traffic via waypoint proxy
+_verify_same snip_verify_the_traffic_via_waypoint_proxy_1 "$snip_verify_the_traffic_via_waypoint_proxy_1_out"
+_verify_same snip_verify_the_traffic_via_waypoint_proxy_2 "$snip_verify_the_traffic_via_waypoint_proxy_2_out"
+
+# Apply Waypoint configuration for 'reviews-svc' service
```
Comment (kfaseela) [inline content/en/docs/ambient/usage/extend-waypoint-wasm/test.sh:88]:
```suggestion
# Apply Waypoint configuration for 'reviews' service
```

File: content/en/docs/ambient/usage/extend-waypoint-wasm/test.sh [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+# apply wasmplugin for a specific service using waypoint
+snip_apply_wasmplugin_for_a_specific_service_using_waypoint_4
+
+# verify the traffic targeting the service
+_verify_same snip_verify_the_traffic_targeting_the_service_1 "$snip_verify_the_traffic_targeting_the_service_1_out"
+_verify_same snip_verify_the_traffic_targeting_the_service_2 "$snip_verify_the_traffic_targeting_the_service_2_out"
+_verify_same snip_verify_the_traffic_targeting_the_service_3 "$snip_verify_the_traffic_targeting_the_service_3_out"
+
+# @cleanup
```
Comment (kfaseela) [inline content/en/docs/ambient/usage/extend-waypoint-wasm/test.sh:103]: why is the snip_cleanup_1 not invoked?

File: content/en/docs/ambient/usage/extend-waypoint-wasm/test.sh [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+snip_uninstall_1
+snip_uninstall_2
+snip_uninstall_3
+samples/bookinfo/platform/kube/cleanup.sh
```
Comment (kfaseela) [inline content/en/docs/ambient/usage/extend-waypoint-wasm/test.sh:107]: can't we use cleanup_bookinfo_sample ?

File: content/en/docs/ambient/usage/extend-waypoint-wasm/test.sh (hunk 1 above)
Comment (adilmp1) [reply content/en/docs/ambient/usage/extend-waypoint-wasm/test.sh:46]: Ok, I'll remove the verifications and add sample applications directly to the Ambient Mesh section

File: content/en/docs/ambient/usage/extend-waypoint-wasm/test.sh (hunk 2 above)
Comment (adilmp1) [reply content/en/docs/ambient/usage/extend-waypoint-wasm/test.sh]: corrected

File: content/en/docs/ambient/usage/extend-waypoint-wasm/test.sh (hunk 3 above)
Comment (adilmp1) [reply content/en/docs/ambient/usage/extend-waypoint-wasm/test.sh:99]: added 👍

**Review by kfaseela — APPROVED** [review]
Thanks @adilmp1

**Pull request merged**